### DISCUSSION TITLE
Can I manually edit data?

### DISCUSSION THREAD

**Discussion by djpalmis** [teslamate-org/teslamate#1946] [Getting started]
I would like to change the first update date: it is the date I installed teslamate, it's not the real firmware update date.

Also: I was charging at home. There was a blackout. Powerline adapter turned off and car was unable to send data to servers.
Turns out that for teslamate that charge was never stopped.. even if it has no data for a long time frame.
So it would be nice to edit that charge session.

**Comment by eliluong** [comment]
You can use something like pgAdmin to connect to the database. You will have to look at the data and study how the database is structured, and then should be able to safely edit it (after making a backup of the database, of course).

**Reply by djpalmis** [reply]
Thank You, I searched on google but honestly I can't do it: It would be better if it were editable on grafana, at least the first firmware update data field.

**Reply by eliluong** [reply]
Unfortunately, I think any incorrect edit can probably ruin the database or prevent data from being parsed properly, and developer would have to implement code for editing, too.

For updates, there is a table called `updates` that has the firmware start and end date and version number:
![image](https://user-images.githubusercontent.com/15806938/132440465-8886ee80-0e51-416a-9d25-f820ed369c1a.png)

For charging, there is a table called `charged_processes` that is the record of each charge start/stop time, along with the data for each charge session (minus the individual battery level, charge voltage, etc at each timepoint, which is stored in a different table).
![image](https://user-images.githubusercontent.com/15806938/132440610-8002a8da-a8d5-4887-8db1-c9a15ad18bf4.png)

I've never edited a charge session before so I don't know what may happen if the end time overlaps with the start time of something else.
